sim/midas/src/main/scala/midas/Config.scala

```diff
@@ -212,3 +222,81 @@ class VitisConfig extends Config(new Config((site, here, up) => {
 class HostDebugFeatures extends Config((site, here, up) => {
   case GenerateTokenIrrevocabilityAssertions => true
 })
+
+class WithWiringTransform extends Config((site, here, up) => {
+  case TargetTransforms => Dependency[firrtl.passes.wiring.WiringTransform] +: up(TargetTransforms, site)
+})
+
+class WithPCIMPorts extends Config((site, here, up) => {
+  case F1ShimHasPCIMPorts => true
+  case FPGAStreamEngineInstantiatorKey =>
+    (e: StreamEngineParameters, p: Parameters) =>
+    new FPGAManagedStreamEngine(p, e)
+  case FPGAManagedAXI4Key   => Some(FPGAManagedAXI4Params(
+    size                = BigInt(1) << 64,  // 128GB (size of Bar 4)
+    dataBits            = 512,                  // 512 bits set by FireSim default
+    idBits              = 6,                    // 6 is a guess; based upon CPUManagedAXI4Params
+    writeTransferSizes  = TransferSizes(512/8), // Default for Firesim
+    readTransferSizes   = TransferSizes(512/8), // Default for Firesim
+    interleavedId       = Some(0)               // Interleaved not currentlky supported therefore 0
+  ))
+})
+
+class EC2F1Config extends Config(
+  new WithPCIMPorts ++
+  new WithWiringTransform ++
+  new F1Config
+)
+
+class WithModelMultiThreading extends Config((site, here, up) => {
+  case midas.EnableModelMultiThreading => true
+})
+
+class WithMultiCycleRams extends Config((site, here, up) => {
+  case midas.GenerateMultiCycleRamModels => true
+})
+
+class MTModels extends WithModelMultiThreading
+class MCRams extends WithMultiCycleRams
```

Comment:
Some of these classes/fields look like duplicates of things in `CompilerConfigs.scala` in `firesim-lib`. Dedup?